cvat/apps/dataset_manager/task.py

```diff
@@ -93,35 +94,47 @@ def add_prefetch_info(cls, queryset):
         ])
         label_qs = JobData.add_prefetch_info(label_qs)
 
+        task_data_queryset = models.Data.objects.all()
+        if prefetch_images:
+            task_data_queryset = task_data_queryset.select_related('video').prefetch_related(
+                Prefetch('images', queryset=models.Image.objects.order_by('frame'))
+            )
+
         return queryset.select_related(
             'segment',
             'segment__task',
         ).prefetch_related(
             'segment__task__project',
             'segment__task__owner',
             'segment__task__assignee',
-            'segment__task__project__owner',
-            'segment__task__project__assignee',
 
-            Prefetch('segment__task__data',
-                queryset=models.Data.objects.select_related('video').prefetch_related(
-                    Prefetch('images', queryset=models.Image.objects.order_by('frame'))
-            )),
+            Prefetch('segment__task__data', queryset=task_data_queryset),
 
             Prefetch('segment__task__label_set', queryset=label_qs),
             Prefetch('segment__task__project__label_set', queryset=label_qs),
         )
 
-    def __init__(self, pk, *, is_prefetched=False, queryset=None):
+    def __init__(
+        self,
+        pk,
+        *,
+        lock_job_in_db: bool = False,
+        queryset: QuerySet | None = None,
+        prefetch_images: bool = False,
+        db_task: models.Task | None = None
+    ):
         if queryset is None:
-            queryset = self.add_prefetch_info(models.Job.objects)
+            if db_task is None:
+                queryset = self.add_prefetch_info(models.Job.objects, prefetch_images=prefetch_images)
+            else:
+                queryset = models.Job.objects.select_related("segment", "segment__task")
```

Comment:
Maybe it's not necessary to request `segment__task` at all in this case? You're going to overwrite it anyway.